app/lib/supergroups_api/supergroup.rb

```diff
@@ -0,0 +1,37 @@
+module SupergroupsApi
```

Comment:
I'd think about renaming this module/class. SupergroupsApi doesn't tell what it does. It's doing a rummager search so perhaps the module or class name should be "Search"?